### PR TITLE
NEPT-1530: Implement varnish path flushing on file delete.

### PR DESCRIPTION
**Pull request by drishu**
### Description

Implement varnish path flushing on file delete.

### Change log

- Added: hook nexteuropa_varnish_file_delete().
- Changed: nexteuropa_varnish_path_delete() to handle file aliases.

### Commands

drush cc all

### CODE REVIEW

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module [hunk 1]
```diff
@@ -279,6 +279,20 @@ function nexteuropa_varnish_node_delete($node) {
   }
 }
 
+/**
+ * Implements hook_file_delete().
+ */
+function nexteuropa_varnish_file_delete($file) {
+  $scheme = file_uri_scheme($file->uri);
+  if ($scheme && file_stream_wrapper_valid_scheme($scheme)) {
```
Comment (drupol) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]:
```suggestion

  if ($scheme !== FALSE && file_stream_wrapper_valid_scheme($scheme) === TRUE) {
```

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module [hunk 2]
```diff
@@ -279,6 +279,20 @@ function nexteuropa_varnish_node_delete($node) {
...
+    // Get path on filesystem.
+    $wrapper = file_stream_wrapper_get_instance_by_scheme($scheme);
+    $dir = $wrapper->getDirectoryPath($file);
+    $paths = array('escaped_alias' => preg_quote($dir . '/' . $file->filename, '/'));
+    _nexteuropa_varnish_purge_paths($paths);
```
Comment (drupol) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]:
```suggestion
    _nexteuropa_varnish_purge_paths(
      array(
        'escaped_alias' => preg_quote(
          $wrapper->getDirectoryPath($file) . '/' . $file->filename,
          '/'
        )
      );
```

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module (hunk 1 above)
Comment (drupol) [reply profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]: Let's give people the opportunity to learn the Drupal API just by reading simple code :-)

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module (hunk 2 above)
Comment (drupol) [reply profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]: There is no need to create local variables.

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module [hunk 3]
```diff
@@ -288,14 +302,15 @@ function nexteuropa_varnish_path_delete($path) {
   }
 
   $node = menu_get_object('node', 1, $path['source']);
+  $is_file = strpos($path['source'], 'file/');
 
-  // Its a node path.
-  if (empty($node)) {
+  // Has to be a node or file path.
+  if (empty($node) && $is_file === FALSE) {
```
Comment (drupol) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]:
```suggestion
  if ($node === NULL && $is_file === FALSE) {
```

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module [hunk 4]
```diff
@@ -288,14 +302,15 @@ function nexteuropa_varnish_path_delete($path) {
...
     return;
   }
 
   // No rule defined for the node: Ignore the rest of the process.
-  if (!nexteuropa_varnish_node_has_rules($node)) {
+  if ($node && !nexteuropa_varnish_node_has_rules($node)) {
```
Comment (drupol) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]:
```suggestion
  if ($node !== NULL && nexteuropa_varnish_node_has_rules($node) === FALSE) {
```

Comment (drupol) [reply profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]: It's either an object or an array, I don't remember.

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module (hunk 3 above)
Comment (drupol) [reply profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]: It's either an array or an object.

File: profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module (hunk 1 above)
Comment (drupol) [inline profiles/common/modules/custom/nexteuropa_varnish/nexteuropa_varnish.module]: Usually before the `if`, you add an empty line to semantically separate the code blocks.

**Review by tomv-be — APPROVED** [review]

**Review by AlbaParrilla — APPROVED** [review]
Ok for me 👍

**Pull request merged**